v20.1/temporary-tables.md

```diff
@@ -0,0 +1,211 @@
+---
+title: Temporary Tables
+summary: CockroachDB supports session-scoped temporary tables.
+toc: true
+---
+
+<span class="version-tag">New in v20.1:</span> CockroachDB supports session-scoped temporary tables (also called "temp tables"). Unlike [persistent tables](create-table.html), temp tables can only be accessed from the session in which they were created, and they are dropped at the end of the session.
+
+CockroachDB also supports [temporary views](views.html#temporary-views) and [temporary sequences](create-sequence.html#temporary-sequences).
+
+{{site.data.alerts.callout_danger}}
+**This is an experimental feature**. The interface and output are subject to change. For details, see the tracking issue [cockroachdb/cockroach#46260](https://github.com/cockroachdb/cockroach/issues/46260).
+{{site.data.alerts.end}}
+
+{{site.data.alerts.callout_info}}
+By default, temp tables are disabled in CockroachDB. To enable temp tables, set the `experimental_enable_temp_tables` [session variable](set-vars.html) to `on`.
+{{site.data.alerts.end}}
+
+## Details
+
+- Temp tables are automatically dropped at the end of the session.
+- A temp table can only be accessed from the session in which it was created.
+- Temp tables persist across transactions in the same session.
+- Persistent tables cannot reference temp tables, but temp tables can reference persistent tables.
+- Temp tables cannot be converted to persistent tables.
+
+## Temporary schemas
+
+Temp tables are not part of the `public` schema. Instead, when you create the first temp table for a session, CockroachDB generates a single temporary schema (`pg_temp_<id>`) for all of the temp tables, [temporary views](views.html#temporary-views), and [temporary sequences](create-sequence.html#temporary-sequences) in the current session for a database. In a session, you can reference the session's temporary schema as `pg_temp`.
+
+## Syntax
+
+To create a temp table, add the following [`TEMP`/`TEMPORARY`](sql-grammar.html#opt_temp_create_table) to the beginning of a [`CREATE TABLE`](create-table.html) or [`CREATE TABLE AS`](create-table-as.html) statement:
```

Comment:
This was a little confusing for me.  I wonder if it is needed at all on this page? Given that the examples show how to do it, and the CREATE TABLE* pages show the syntax diagrams I might argue it's not necessary, and we could refer the user to the CREATE TABLE* pages if they care about syntax diagrams.